docs/src/arrays.md

```diff
@@ -0,0 +1,800 @@
+    ∂s = eltype(A) <: Real ? Zero() : im * imag(b) * s
+    # tangents of tuples are of type Composite{<:Tuple}
+    ∂Ω = Composite{typeof(Ω)}(∂l, ∂s)
+    return (Ω, ∂Ω)
+end
+```
+
+### Reverse-mode rule
+
+```math
+\begin{align*}
+&\operatorname{real}\left( \operatorname{tr}\left(
+    \overline{l}^H \dot{l}
+\right) \right) +
+\operatorname{real}\left( \operatorname{tr}\left(
+    \overline{s}^H \dot{s}
+\right) \right)
+    && \text{left-hand side of } \eqref{pbidentmat}\\
+&= \operatorname{real}\left( 
+       \operatorname{conj} \left( \overline{l} \right) \dot{l} +
+       \operatorname{conj} \left( \overline{s} \right) \dot{s}
+   \right) \\
+&= \operatorname{real}\left( 
+       \operatorname{conj} \left( \overline{l} \right) \operatorname{real}(b) +
+       i \operatorname{conj} \left( \overline{s} \right) s \operatorname{imag}(b)
+   \right)
+       && \text{substitute } \eqref{logabsdet_ldot} \text{ and } \eqref{logabsdet_sdot} \\
+&= \operatorname{real}\left( 
+       \operatorname{real}\left( \overline{l} \right) \operatorname{real}(b) -
+       \operatorname{imag} \left(
+           \operatorname{conj} \left( \overline{s} \right) s
+       \right) \operatorname{imag}(b)
+   \right)
+       && \text{discard imaginary parts} \\
+&= \operatorname{real}\left(
+       \left(
+           \operatorname{real} \left( \overline{l} \right) +
+           i \operatorname{imag} \left(
+               \operatorname{conj} \left( \overline{s} \right) s
+           \right)
+       \right) b
+   \right)
+       && \text{gather parts of } b \\
+&= \operatorname{real}\left(
+       \left(
+           \operatorname{real} \left( \overline{l} \right) +
+           i \operatorname{imag} \left(
+               \operatorname{conj} \left( \overline{s} \right) s
+           \right)
+       \right)
+       \operatorname{tr}(A^{-1} \dot{A})
+   \right)
+       && \text{substitute } b \text{ from } \eqref{logabsdet_b} \\
+&= \operatorname{real}\left( \operatorname{tr} \left(
+       \left(
+           \operatorname{real} \left( \overline{l} \right) +
+           i \operatorname{imag} \left(
+               \operatorname{conj} \left( \overline{s} \right) s
+           \right)
+       \right)
+       A^{-1} \dot{A}
+   \right) \right)
+       && \text{bring scalar within } \operatorname{tr} \\
+&= \operatorname{real}\left( \operatorname{tr} \left( \overline{A}^H \dot{A} \right) \right)
+       && \text{right-hand side of } \eqref{pbidentmat}\\
+\end{align*}
+```
+
+Now we solve for $\overline{A}$:
+
+```math
+\begin{align*}
+\overline{A} &= \left( \left(
+    \operatorname{real} \left( \overline{l} \right) +
+    i \operatorname{imag} \left( \operatorname{conj} \left( \overline{s} \right) s \right)
+\right) A^{-1} \right)^H\\
+&= \left(
+    \operatorname{real} \left( \overline{l} \right) +
+    i \operatorname{imag} \left( \operatorname{conj} \left( s \right) \overline{s} \right)
+\right) A^{-H}
+\end{align*}
+```
+
+!!! note
```

Comment:
tiny things, but i wonder is this note can be moved to after the `rrule` code below? Since it discusses both pushforwards and pullbacks, it kinda breaks the flow of the section on Reverse-mode rule with where it is now

Reply:
Good point, thanks. Done!